src/Components/Blazor/WebAssembly.Authentication/src/Services/AccessTokenResult.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// Represents the result of trying to provision an access token.
+    /// </summary>
+    public class AccessTokenResult
+    {
+        /// <summary>
+        /// Gets or sets the status of the current operation. See <see cref="AccessTokenResultStatus"/> for a list of statuses.
+        /// </summary>
+        public string Status { get; set; }
```

Comment:
Some concerns with this API:

First, is it really true that the only two possible statuses are "success" and "redirection required"? Isn't it possible to get a result like "access denied" if you're asking for a scope not grantable to this user? Does that case surface as an exception maybe?

Second, would it be possible to avoid having `Status` be of type `string`? Doing it this way means the developer has to go and read docs to find out all the possible statuses. In their code, they have to check for all possible statuses (and hardcode their names in their code) because they can't be sure they can interpret the meanings of the other properties without doing so. I know there are only two statuses, but the developer doesn't know that. Maybe they fear that other statuses will be added in the future, so they even have to check for unknown status names and throw in that case.

Instead, if the outcome really is boolean, why not replace `string Status` with `boolean Success`?

Reply:
I would even consider not having a `token` property on this type, and instead forcing developers to obtain the token using an API like this:

```
if (result.TryGetToken(out var token))
{
    // Use token
}
else
{
    // Use result.RedirectionUrl
}
```

Another possibility:

```cs
if (result.TryGetTokenOrRedirect(out var token))
{
    // Use token (and if a redirection happens, we don't enter this branch)
}
```

Reply:
> and hardcode their names in their code

Update: I realise that's not fully true since people can use the `Status.PropertyName` constants, but it's not obvious that such a set of constants exists.

Reply:
I've updated this to add the method you suggested for ease of use.

Reply:
I've updated the pattern here.

Reply:
Any sort of failure that happens here we will subsume it under redirect, so that you go to the redirect url and then see the failure in the appropriate context.